README.md

```diff
@@ -200,6 +200,46 @@ Or, to format only specified files, use
 nox -s format -- example.py pygdbmi/IoManager.py
 ```
 
+### Making a release
+
+Only maintainers of the [pygdbmi package on PyPi](https://pypi.org/project/pygdbmi/) can make a release.
+
+In the following steps, replace these strings with the correct values:
+
+- `<REMOTE>` is the name of the remote for the main pygdbmi repository (for instance, `origin`)
+- `<VERSION>` is the version number chosen in step 2.
+
+To make a release:
+
+1. Checkout the `master` branch and pull from the main repository with `git pull <REMOTE> master`
+2. Decide the version number for the new release:
+   - If making a breaking change, increment the second component of the version number
+   - Otherwise, increment the latest component
+3. Update `CHANGELOG.md` to list the chosen version number instead of `## dev`
+4. Update `__version__` in `pygdbmi/__init__.py` to the chosen version number
+5. Commit you changes, for instance using `git commit -a -m 'Bump version to <VERSION> for release'`
+6. Check that the docs look fine by serving them locally with `nox -s serve_docs`
+7. You are now ready to make a release, but first verify everything is fine:
+   - `nox -s lint`
+   - `nox -s tests`
+8. Push your changes straight to `master` with `git push <REMOTE> master`
+9. Tag the release with `git tag <VERSION>`
+10. Push the tag with `git tag <REMOTE> <VERSION>`
+11. Publish the release with `nox -s publish`
+12. Publish the docs with `nox -s publish_docs`
+13. Verify that the [PyPi page for pygdbmi](https://pypi.org/project/pygdbmi/) looks correct
+14. Verify that the [published docs](https://cs01.github.io/pygdbmi/) look correct
+15. Prepare for changes for the next release by adding something like this above the previous entries in `CHANGELOG.md`:
+
+    ```
+    ## dev
+
+    - *Replace this line with new entries*
+    ```
+
+16. Commit the change with `git commit -m 'Prepare for work on the next release' CHANGELOG.md`
```

Comment:
suggest modifying the version here to have the `.dev0` suffix (but not publishing it to PyPI)

https://peps.python.org/pep-0440/#public-version-identifiers
https://peps.python.org/pep-0440/#developmental-releases